pgx-pg-config/src/lib.rs

```diff
@@ -528,3 +527,35 @@ fn does_db_exist(pg_config: &PgConfig, dbname: &str) -> eyre::Result<bool> {
         Ok(count > 0)
     }
 }
+
+#[test]
+fn parse_version() {
+    // Check some valid version strings
+    let versions = [
+        ("PostgreSQL 10.22", 10, 22),
+        ("PostgreSQL 11.2", 11, 2),
+        ("PostgreSQL 11.17", 11, 17),
+        ("PostgreSQL 12.12", 12, 12),
+        ("PostgreSQL 13.8", 13, 8),
+        ("PostgreSQL 14.5", 14, 5),
+        ("PostgreSQL 11.2-FOO-BAR+", 11, 2),
+        ("PostgreSQL 10.22-", 10, 22),
+    ];
+    for (s, major_expected, minor_expected) in versions {
+        let (major, minor) =
+            PgConfig::parse_version_str(s).expect("Unable to parse version string");
+        assert_eq!(major, major_expected, "Major varsion should match");
+        assert_eq!(minor, minor_expected, "Minor version should match");
+    }
+
+    // Check some invalid version strings
+    let _ = PgConfig::parse_version_str("10.22").expect_err("Parsed invalid version string");
+    let _ =
+        PgConfig::parse_version_str("PostgresSQL 10").expect_err("Parsed invalid version string");
```

Comment:
...are we entirely sure this one should be invalid?

Reply:
What should it come out as?

Reply:
@workingjubilee I do not know. I just matched the behavior of the existing code. Currently "PostgresSQL 10" will fail with an invalid minor version error. Happy to change it if you feel this should really be interpreted as 10.0 or something though.